### PR TITLE
tests: Automated tests for things that require user input

### PR DESCRIPTION
**Pull request by kmeisthax**
This PR adds a mechanism for adding canned user input to automated tests, to allow testing things that ordinarily require user interaction, such as clicking on a button.

The input format added here is identical to the one that I used in [FlashTAS](https://github.com/kmeisthax/flashtas), and currently only encompasses mouse input, though it could certainly be extended to support keyboards, touch, and so on. It is added as a new separate crate in `tests/input-format`, so that FlashTAS can continue to use it as we expand the format.

The test authoring experience for input in tests would work as follows:

 * Create your test as per usual.
 * Craft an input file for the test and store it next to the SWF as `input.json`.
 * Run the SWF and the JSON file in FlashTAS and record the output in Scout CC (side note: does anyone know how to get the tracelog without using Scout or Flash Pro?)
 * Add the test to Ruffle as usual with the FlashTAS-derived `output.txt`, and hack on Ruffle until tests pass

### CODE REVIEW

**Comment by kmeisthax** [comment]
Apologies for this taking so long. I wanted to include at least one sample test with an input file, but had problems getting it to work in FlashTAS to generate comparison output. It turns out that AS2 buttons have some interesting input handling quirks, where on Windows they don't register multiple button presses unless you send a mouse movement (`WM_MOUSEMOVE`) in between each click (`WM_LBUTTONDOWN` / `WM_LBUTTONUP`). I genuinely thought FlashTAS was broken, and spent more time than I care to admit spying on the message pump to try and uncover some oversight that didn't actually exist.

Once this is merged in I can start writing mouse input tests for #5876 which should hopefully unblock the problems it still has on Parameters.

File: tests/tests/swfs/avm1/mouse_events/test.html
Comment (Herschel) [inline tests/tests/swfs/avm1/mouse_events/test.html:15]: Mistakenly added this HTML file?

File: tests/input-format/src/format.rs
Comment (Herschel) [inline tests/input-format/src/format.rs:47]: TODO maybe someday: It'd be nice if we could reuse some of the `core` types here. At least things like `MouseButton` and `KeyCode`, or possibly this could be all of `PlayerEvent`.

Naturally I wouldn't want to require all of `ruffle_core` here for a few tiny types, so maybe we can keep this in mind if we ever reorganize our workspace structure. On the other hand, having the separation might be a bit nicer to prevent the input format breakage if we add things to the `core` side structs.

**Review by Herschel — APPROVED** [review]
Thank you! I'm excited to make some cool tests for all the wacky button behavior. :-)

Could you remove the extra HTML file?

File: tests/tests/swfs/avm1/mouse_events/test.html
Comment (kmeisthax) [reply tests/tests/swfs/avm1/mouse_events/test.html]: Yup, this was a mistake and it's been removed in a rebase.

File: tests/input-format/src/format.rs
Comment (kmeisthax) [reply tests/input-format/src/format.rs:47]: There's a few subtle things that need to be different from `PlayerEvent`, too; while it might be justifiable to add `serde` impls for it, I doubt a `Wait` event would make sense outside of automated contexts.

**Pull request merged**
